### DISCUSSION TITLE
Accessing useSwiper() equivalent using React Web Components

### DISCUSSION THREAD

**Discussion by cojoclaudiu** [nolimits4web/swiper#6864] [Q&A]
Hello,

I'm trying to switch to Swiper Element in React following this tutorial https://swiperjs.com/blog/using-swiper-element-in-react but I would like to know how I can access swiper instance inside the children components.

I should create a context provider?

For example I have some custom navigation buttons where I would like to use swiper.slideNext(), swiper.slidePrev() and show the swiper activeIndex when the image is changed.

**Comment by kirgans** [comment]
Attach a ref to your parent swiper and pass the ref down to your children comp, it should work.

**Reply by cojoclaudiu** [reply]
I have a wrapper component over the swiper container, I don't see how I can pass the ref to the children without context provider. And I've tried that but the ref doesn't trigger a re-render when I change the slide and for example I would like to display the active index 
```ts
function Swiper(
	props: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement> & SwiperProps, HTMLElement>
) {
	const { children, ...rest } = props;
	const swiperRef = useRef<SwiperContainer>(null);

	useEffect(() => {
		// Register Swiper web component
		register();

		if (swiperRef.current) {
			// pass component props to parameters
			const params = {
				...rest,
			};
			// Assign it to swiper element
			Object.assign(swiperRef.current, params);

			// initialize swiper
			swiperRef.current.initialize();
		}
	}, [swiperRef, rest]);

	return (
		<swiper-container
			//@ts-expect-error because is a web component and we need to pass value as a string
			init='false'
			ref={swiperRef}
		>
			{children}
		</swiper-container>
	);
}
```

**Reply by kirgans** [reply]
seems to work like this ? 
https://codesandbox.io/p/sandbox/nested-swipers-virtual-slides-forked-ttwm7x?file=%2Fapp%2Fpage.tsx%3A14%2C22

**Reply by kirgans** [reply]
in addition, don't pass ref in the array of deps of the useEffect
https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

**Reply by cojoclaudiu** [reply]
@spacedev75 thank you, I will give a try

**Reply by cojoclaudiu** [reply]
@spacedev75 I get this error Uncaught TypeError: forwardedRef.current.initialize is not a function
